### PR TITLE
Stop tests route from blocking the event loop with sync fs methods

### PR DESCRIPTION
**Pull request by JakeChampion**
This might be easier to review by ignoring whitespace and shrinking the size of tabs --> https://github.com/Financial-Times/polyfill-service/pull/1144/files?w=1&ts=2

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ function listAllPolyfills() {
 }
 
 function describePolyfill(featureName) {
-	return sourceslib.getPolyfill(featureName);
+	return sourceslib.getPolyfillMetaSync(featureName);
```
Comment (triblondon) [inline lib/index.js]: Is this a breaking change?  Previously returned a promise?

File: service/RumReport.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ function Perf(rawopts) {
 	;
 
 	const sqlQuery = `SELECT ${sqlFields} FROM requests WHERE req_time BETWEEN (CURDATE() - INTERVAL ${options.period} DAY) AND CURDATE() AND data_center IS NOT NULL AND perf_req IS NOT NULL LIMIT 1000000`;
-console.log(options.minSample);
+	console.log(options.minSample);
```
Comment (triblondon) [inline service/RumReport.js]: Hm, this is probably accidental debug that we should remove

File: service/RumReport.js [hunk 3]
```diff
@@ -151,9 +151,8 @@ function Compat() {
 		})
 		.then(results => {
 			return dbconn.end().then(() => {
-				return Promise.all(results[0].map(rec => {
-					return polyfillio.describePolyfill(rec.feature_name)
-					.then(polyfill => {
+				return results[0].map(rec => {
+					const polyfill = polyfillio.describePolyfill(rec.feature_name);
```
Comment (triblondon) [inline service/RumReport.js]: Should this be using the sources lib directly rather than the external API, I wonder?

File: service/routes/test.js [hunk 4]
```diff
@@ -40,30 +42,32 @@ function createEndpoint(type, polyfillio) {
...
+							const baseDir = path.join(__dirname, '../../polyfills');
+							const testFile = path.join(baseDir, config.baseDir, '/tests.js');
+							return readFile(testFile)
+								.then(tests => {
+									return {
+										feature: featureName,
+										detect: config.detectSource ? config.detectSource : false,
+										tests
+									};
+								});
```
Comment (triblondon) [inline service/routes/test.js]: Move semicolon to next line to avoid jumping two indent levels?

File: tasks/node/buildsources.js [hunk 5]
```diff
@@ -233,14 +239,15 @@ Promise.resolve()
         .map(absolute => new Polyfill(absolute, path.relative(src, absolute)))
         .filter(polyfill => polyfill.hasConfigFile)
         .map(polyfill => polyfill.loadConfig()
-            .then(() => polyfill.loadSources())
+			.then(() => polyfill.loadSources())
+			.then(() => polyfill.updateConfig())
```
Comment (triblondon) [inline tasks/node/buildsources.js]: Are you sure your indentation is right here?

File: service/routes/test.js [hunk 6]
```diff
@@ -40,30 +42,32 @@ function createEndpoint(type, polyfillio) {
...
 				return Promise.all(featuresList.map(featureName => {
-					return polyfillio.describePolyfill(featureName).then(config => ({[featureName]: config}) );
-				})).then(featureObjs => Object.assign({}, ...featureObjs));
+					return Promise.resolve()
+						.then(() => {
+
+						const config = polyfillio.describePolyfill(featureName);
+
+						const isTestable = config.isTestable;
+						const isPublic = config.isPublic;
```
Comment (triblondon) [inline service/routes/test.js]: Maybe no need to reassign these anymore?  It now feels inconsistent with the treatment of `config.hasTests`

File: tasks/node/buildsources.js [hunk 7]
```diff
@@ -115,13 +119,15 @@ class Polyfill {
             .then(data => {
                 this.config = JSON.parse(data);
                 this.config.detectSource = '';
-                this.config.baseDir = this.path.relative;
+				this.config.baseDir = this.path.relative;
```
Comment (rowanmanning) [inline tasks/node/buildsources.js]: A few tab/space indent inconsistencies in this file

File: lib/index.js (hunk 1 above)
Comment (JakeChampion) [reply lib/index.js]: Yes, removing `getPolyfill` from `lib/sources.js` is also a breaking change.

File: tasks/node/buildsources.js (hunk 5 above)
Comment (JakeChampion) [reply tasks/node/buildsources.js]: I am sure, seems the files was in spaces :-(

File: tasks/node/buildsources.js (hunk 7 above)
Comment (JakeChampion) [reply tasks/node/buildsources.js]: Seems the files was using spaces for indentation

File: lib/index.js (hunk 1 above)
Comment (JakeChampion) [reply lib/index.js]: I made it a non-breaking change now

**Review by rowanmanning — APPROVED** [review]
Looks great 👍

**Pull request merged**